src/main/java/com/linecorp/armeria/client/metrics/MetricCollectingClient.java

```diff
@@ -0,0 +1,39 @@
+package com.linecorp.armeria.client.metrics;
+
+import java.util.function.Function;
+
+import com.codahale.metrics.MetricRegistry;
+
+import com.linecorp.armeria.client.Client;
+import com.linecorp.armeria.common.metrics.DropwizardMetricConsumer;
+
+
+public final class MetricCollectingClient {
+
+    /**
+     * A {@link Client} decorator that tracks request stats using the Dropwizard metrics library.
+     * To use, simply prepare a {@link MetricRegistry} and add this decorator to a client.
+     *
+     * <p>Example:
+     * <pre>{@code
+     * MetricRegistry metricRegistry = new MetricRegistry();
+     * MyService.Iface client = new ClientBuilder(uri)
+     *     .decorate(MetricCollectingClient.newDropwizardMetricsDecorator("MyService", metricRegistry))
+     *     .build(MyService.Iface.class);
+     * }
+     * </pre>
+     * </p>
+     * <p>It is generally recommended to define your own name for the service instead of using something like
+     * the Java class to make sure otherwise safe changes like renames don't break metrics.</p>
+     */
+    public static Function<Client, Client> newDropwizardMetricsDecorator(
```

Comment:
Went with newDropwizardDecorator, but in practice maybe there won't be any other implementations.